<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ol.lst-kix_gd110v9f4dxg-0.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-0 0}.lst-kix_gd110v9f4dxg-7>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-7}ol.lst-kix_ej4ibi9ebo3s-4.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-4 0}ol.lst-kix_gd110v9f4dxg-7.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-7 0}.lst-kix_gd110v9f4dxg-1>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-1}.lst-kix_ej4ibi9ebo3s-5>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-5,lower-roman) ". "}ol.lst-kix_ej4ibi9ebo3s-7.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-7 0}ol.lst-kix_gd110v9f4dxg-3.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-3 0}ol.lst-kix_ej4ibi9ebo3s-1.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-1 0}.lst-kix_ej4ibi9ebo3s-6>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-6,decimal) ". "}ol.lst-kix_gd110v9f4dxg-4.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-4 0}.lst-kix_ej4ibi9ebo3s-4>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-4}.lst-kix_ej4ibi9ebo3s-7>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-7,lower-latin) ". "}.lst-kix_ej4ibi9ebo3s-8>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-8,lower-roman) ". "}.lst-kix_gd110v9f4dxg-0>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-0}.lst-kix_gd110v9f4dxg-2>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-2,lower-roman) ". "}.lst-kix_gd110v9f4dxg-4>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-4,lower-latin) ". "}ol.lst-kix_gd110v9f4dxg-8{list-style-type:none}.lst-kix_gd110v9f4dxg-3>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-3,decimal) ". "}.lst-kix_gd110v9f4dxg-6>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-6,decimal) ". "}ol.lst-kix_gd110v9f4dxg-8.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-8 0}ol.lst-kix_ej4ibi9ebo3s-3.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-3 0}.lst-kix_gd110v9f4dxg-5>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-5,lower-roman) ". "}.lst-kix_gd110v9f4dxg-2>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-2}.lst-kix_ej4ibi9ebo3s-0>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-0}.lst-kix_ej4ibi9ebo3s-3>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-3}ol.lst-kix_ej4ibi9ebo3s-0.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-0 0}.lst-kix_ej4ibi9ebo3s-6>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-6}.lst-kix_gd110v9f4dxg-8>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-8,lower-roman) ". "}.lst-kix_gd110v9f4dxg-8>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-8}.lst-kix_gd110v9f4dxg-7>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-7,lower-latin) ". "}ol.lst-kix_gd110v9f4dxg-2.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-2 0}ol.lst-kix_gd110v9f4dxg-4{list-style-type:none}ol.lst-kix_gd110v9f4dxg-5{list-style-type:none}ol.lst-kix_gd110v9f4dxg-6{list-style-type:none}ol.lst-kix_gd110v9f4dxg-7{list-style-type:none}ol.lst-kix_gd110v9f4dxg-0{list-style-type:none}ol.lst-kix_gd110v9f4dxg-1{list-style-type:none}ol.lst-kix_gd110v9f4dxg-5.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-5 0}.lst-kix_gd110v9f4dxg-5>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-5}ol.lst-kix_gd110v9f4dxg-2{list-style-type:none}ol.lst-kix_gd110v9f4dxg-3{list-style-type:none}.lst-kix_gd110v9f4dxg-4>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-4}ol.lst-kix_ej4ibi9ebo3s-2.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-2 0}ol.lst-kix_ej4ibi9ebo3s-6.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-6 0}.lst-kix_ej4ibi9ebo3s-7>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-7}.lst-kix_ej4ibi9ebo3s-1>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-1}.lst-kix_gd110v9f4dxg-1>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-1,lower-latin) ". "}ol.lst-kix_gd110v9f4dxg-1.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-1 0}ol.lst-kix_gd110v9f4dxg-6.start{counter-reset:lst-ctn-kix_gd110v9f4dxg-6 0}ol.lst-kix_ej4ibi9ebo3s-6{list-style-type:none}ol.lst-kix_ej4ibi9ebo3s-5{list-style-type:none}ol.lst-kix_ej4ibi9ebo3s-8{list-style-type:none}.lst-kix_ej4ibi9ebo3s-8>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-8}ol.lst-kix_ej4ibi9ebo3s-7{list-style-type:none}.lst-kix_gd110v9f4dxg-0>li:before{content:"" counter(lst-ctn-kix_gd110v9f4dxg-0,decimal) ". "}.lst-kix_ej4ibi9ebo3s-2>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-2}ol.lst-kix_ej4ibi9ebo3s-8.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-8 0}.lst-kix_gd110v9f4dxg-3>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-3}.lst-kix_gd110v9f4dxg-6>li{counter-increment:lst-ctn-kix_gd110v9f4dxg-6}ol.lst-kix_ej4ibi9ebo3s-0{list-style-type:none}ol.lst-kix_ej4ibi9ebo3s-2{list-style-type:none}.lst-kix_ej4ibi9ebo3s-5>li{counter-increment:lst-ctn-kix_ej4ibi9ebo3s-5}ol.lst-kix_ej4ibi9ebo3s-1{list-style-type:none}ol.lst-kix_ej4ibi9ebo3s-4{list-style-type:none}ol.lst-kix_ej4ibi9ebo3s-3{list-style-type:none}.lst-kix_ej4ibi9ebo3s-3>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-3,decimal) ". "}.lst-kix_ej4ibi9ebo3s-4>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-4,lower-latin) ". "}.lst-kix_ej4ibi9ebo3s-2>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-2,lower-roman) ". "}ol.lst-kix_ej4ibi9ebo3s-5.start{counter-reset:lst-ctn-kix_ej4ibi9ebo3s-5 0}.lst-kix_ej4ibi9ebo3s-0>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-0,decimal) ". "}.lst-kix_ej4ibi9ebo3s-1>li:before{content:"" counter(lst-ctn-kix_ej4ibi9ebo3s-1,lower-latin) ". "}ol{margin:0;padding:0}table td,table th{padding:0}.c28{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:1pt;border-right-width:1pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:117pt;border-top-color:#000000;border-bottom-style:solid}.c22{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:1pt;border-right-width:1pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:1pt;border-top-style:solid;border-left-style:solid;border-bottom-width:1pt;width:156pt;border-top-color:#000000;border-bottom-style:solid}.c18{-webkit-text-decoration-skip:none;color:#1155cc;font-weight:400;text-decoration:underline;vertical-align:baseline;text-decoration-skip-ink:none;font-size:11pt;font-family:"Arial";font-style:normal}.c12{padding-top:18pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:justify;height:16pt}.c5{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c0{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:italic}.c29{padding-top:12pt;padding-bottom:6pt;line-height:1.0;page-break-after:avoid;orphans:2;widows:2;text-align:justify}.c32{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:20pt;font-family:"Arial";font-style:normal}.c23{padding-top:0pt;padding-bottom:16pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c34{padding-top:12pt;padding-bottom:4pt;line-height:1.0;page-break-after:avoid;orphans:2;widows:2;text-align:justify}.c19{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:16pt;font-family:"Arial";font-style:normal}.c42{padding-top:12pt;padding-bottom:4pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:justify}.c9{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c39{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:italic}.c6{color:#666666;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:15pt;font-family:"Arial";font-style:normal}.c8{padding-top:20pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:justify}.c30{padding-top:0pt;padding-bottom:3pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:justify}.c4{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:10pt;font-family:"Arial";font-style:italic}.c11{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left;height:11pt}.c41{padding-top:18pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c38{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:26pt;font-family:"Arial";font-style:normal}.c7{color:#434343;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:14pt;font-family:"Arial";font-style:normal}.c1{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:justify;height:11pt}.c27{padding-top:12pt;padding-bottom:0pt;line-height:1.0;orphans:2;widows:2;text-align:justify}.c50{padding-top:0pt;padding-bottom:16pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c14{padding-top:12pt;padding-bottom:7pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c35{padding-top:0pt;padding-bottom:12pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c52{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:right}.c20{padding-top:12pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c15{padding-top:12pt;padding-bottom:12pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c17{color:#000000;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c47{color:#434343;text-decoration:none;vertical-align:baseline;font-size:14pt;font-family:"Arial";font-style:normal}.c2{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c37{padding-top:16pt;padding-bottom:4pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c24{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c53{padding-top:18pt;padding-bottom:6pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c49{color:#000000;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial"}.c10{margin-left:18pt;padding-top:3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c45{border-spacing:0;border-collapse:collapse;margin-right:auto}.c25{padding-top:3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c16{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c21{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c46{max-width:468pt;padding:72pt 72pt 72pt 72pt}.c3{color:inherit;text-decoration:inherit}.c33{background-color:#ffffff}.c13{font-weight:700}.c36{margin-left:36pt}.c40{height:14pt}.c48{font-style:italic}.c44{height:20pt}.c43{page-break-after:avoid}.c31{height:0pt}.c26{height:11pt}.c51{font-size:12pt}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c33 c46 doc-content"><p class="c1"><span class="c5"></span></p><p class="c30 title" id="h.f4ssjdz1bn7c"><span class="c38">Developing an Integrated Pipeline for the Analysis and Visualization of Single-Cell RNA Sequencing Datasets</span></p><p class="c11"><span class="c5"></span></p><p class="c23 subtitle" id="h.b50ml62y4ruy"><span class="c6">Machine Learning in Computational Biology: Assignment #3</span></p><p class="c23 subtitle" id="h.elsrbup9ueiv"><span class="c6">Marina Thalassini Philippidou 7115152200022</span></p><p class="c23 subtitle" id="h.4durvhzdf4x2"><span class="c6">2/6/2023</span></p><p class="c11"><span class="c5"></span></p><p class="c25"><span class="c17 c13"><a class="c3" href="#h.qcq06gv4jesy">Abstract&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1</a></span></p><p class="c25"><span class="c17 c13"><a class="c3" href="#h.q2ljbnwfxm0b">Introduction&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2</a></span></p><p class="c25"><span class="c17 c13"><a class="c3" href="#h.ttp4wu5chtve">Data &amp; Methods&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.8q9j7fw35z3r">Data and Preprocessing&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.scnm8v87nt4z">Dimensionality Reduction&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.bmzkptaywdj">PCA-UMAP-t-SNE&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.1xamszrfpmhy">2D and 3D visualization&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.vol9joc5bvfd">Optimizing UMAP using k-means clustering&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;6</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.qrzyuvh26txo">Optimizing PCA using cumulative variance&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;9</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.ip1s8pv7zft8">Clustering&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.ohf86kq0qg7r">Optimize GMM model for PCA and UMAP results&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.bzno4s8exycs">Results&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;11</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.marn1487nnra">GMM model for PCA Data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;11</a></span></p><p class="c25 c36"><span class="c5"><a class="c3" href="#h.na7jg3pmonxv">GMM model for UMAP Data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;14</a></span></p><p class="c25"><span class="c17 c13"><a class="c3" href="#h.u9ojcp24j8nj">Discussion&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;17</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.4xzz08uujull">Bonus Research&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;18</a></span></p><p class="c10"><span class="c5"><a class="c3" href="#h.1a5qe6g1y5tr">Results&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;18</a></span></p><p class="c25"><span class="c17 c13"><a class="c3" href="#h.bjb2tc14l8xd">References&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;19</a></span></p><p class="c11"><span class="c5"></span></p><h1 class="c8" id="h.qcq06gv4jesy"><span class="c32">Abstract</span></h1><p class="c2"><span class="c5">Single-cell RNA sequencing (scRNA-seq) has revolutionized genomics and molecular biology, enabling researchers to study gene expression at the single-cell level. It has broadened our understanding of cellular heterogeneity, developmental processes, disease progression, and various biological phenomena. It is a very important tool in biology because by using it we can study Cellular Heterogeneity to identify rare cell population, Cell Development and Lineage Tracing by unraveling cellular differentiation pathways and lineage relationships . We can also &nbsp; perform more accurate Disease Research by identifying DNA variations which are related to diseases such as cancer, neurodegenerative disorders, and immune-related conditions and finding the specific biomarkers characterizing each disease. </span></p><p class="c2"><span>While scRNA-seq offers numerous advantages, it is important to acknowledge that these advantages come with certain challenges. One of them is Data Dimensionality, scRNA-seq generates large and high-dimensional datasets, necessitating sophisticated computational techniques for data analysis, visualization, and interpretation. </span><span class="c33">Another one stems from the fact that technical variability in scRNA-seq protocols can introduce biases and noise, requiring proper normalization and quality control methods to mitigate these effects. Additionally, scRNA-seq data often exhibits sparsity, with many genes being undetected or expressed at low levels in individual cells. This sparsity, along with dropout events (where lowly expressed genes go undetected), poses challenges for accurate data analysis and interpretation. Data Integration is yet another challenge to face since technical differences, batch effects, and heterogeneity across datasets pose difficulties in harmonizing and integrating data effectively. &nbsp; Furthermore, </span><span class="c5">analyzing scRNA-seq data demands significant computational resources, and the experimental costs associated with sample preparation and sequencing can be prohibitive for some researchers.</span></p><p class="c2"><span>In this paper, we address some of the challenges associated with scRNA-seq data analysis by focusing on optimizing dimensionality reduction algorithms and exploring different analysis pipelines. Specifically, we aim to enhance the analysis of scRNA-seq datasets by refining algorithms such as principal component analysis (PCA) and </span><span class="c48">t</span><span class="c5">-distributed stochastic neighbor embedding (t-SNE) and Uniform Manifold Approximation and Projection (UMAP) to effectively handle the high-dimensional nature of the data. Additionally, we experiment with various analysis pipelines, incorporating steps such as data preprocessing, normalization, cell clustering, and differential gene expression analysis. By evaluating and comparing these approaches, we aim to identify the most suitable strategies for robust and accurate scRNA-seq data analysis. Through this research, we strive to contribute to the advancement of understanding cellular heterogeneity and uncovering underlying biological mechanisms</span></p><p class="c1"><span class="c5"></span></p><p class="c43 c26 c50 subtitle" id="h.bnybn4iv7cre"><span class="c6"></span></p><h1 class="c8" id="h.q2ljbnwfxm0b"><span class="c32">Introduction</span></h1><p class="c35"><span class="c5">Single-cell RNA sequencing (scRNA-seq) has revolutionized the field of genomics, enabling the characterization of gene expression at the single-cell level. However, the analysis of scRNA-seq datasets presents significant challenges due to the high dimensionality, technical variability, and sparsity of the data. In this paper, we present an integrated pipeline designed to tackle these challenges and provide a comprehensive approach for analyzing and visualizing scRNA-seq data. Our pipeline combines dimensionality reduction methods, optimization of a Gaussian Mixture Model (GMM) algorithm, and advanced data visualization techniques to extract meaningful insights and drive conclusive results from the data.</span></p><p class="c15"><span class="c5">The primary objective of our study is to develop a robust pipeline that integrates state-of-the-art dimensionality reduction techniques into the analysis workflow. We leverage methods such as Principal Component Analysis (PCA), t-Distributed Stochastic Neighbor Embedding (t-SNE), and Uniform Manifold Approximation and Projection (UMAP) to reduce the high-dimensional scRNA-seq data into lower-dimensional representations that preserve the underlying structure and relationships of the cells.</span></p><p class="c15"><span class="c5">Furthermore, we focus on optimizing a Gaussian Mixture Model algorithm specifically tailored for scRNA-seq data analysis. By optimizing the model&#39;s parameters and incorporating prior biological knowledge, we aim to improve the accuracy of cell clustering and identification of distinct cell populations within the dataset.</span></p><p class="c15"><span>In addition to data analysis, visualization plays a vital role in interpreting scRNA-seq results. Hence, our pipeline includes advanced visualization techniques to depict the cellular heterogeneity, gene expression patterns, and developmental trajectories within the dataset. By providing intuitive visual representations, we facilitate the exploration and understanding of complex scRNA-seq data, ultimately </span><span>reaching </span><span class="c5">&nbsp;meaningful conclusions.</span></p><p class="c15"><span>Through the integration of dimensionality reduction, optimization of the GMM algorithm, and advanced data visualization, our pipeline offers a comprehensive solution for analyzing and interpreting scRNA-seq data. By applying this pipeline to a real-world scRNA-seq dataset, we demonstrate its effectiveness in uncovering biological insights and </span><span>obtaining </span><span class="c5">conclusive findings. Our study contributes to the advancement of scRNA-seq analysis methodologies and enables researchers to gain a deeper understanding of cellular heterogeneity and gene expression dynamics.</span></p><p class="c15"><span class="c13">Keywords</span><span class="c5">: single-cell RNA sequencing, scRNA-seq, analysis pipeline, dimensionality reduction, Gaussian Mixture Model, visualization, cellular heterogeneity, gene expression dynamics.</span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><h1 class="c8" id="h.ttp4wu5chtve"><span class="c32">Data &amp; Methods</span></h1><p class="c2"><span class="c5">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br></span></p><h2 class="c29" id="h.8q9j7fw35z3r"><span class="c19">Data and Preprocessing</span></h2><p class="c27"><span class="c13"><br></span><span>We work with an already curated dataset provided from our institution. O</span><span>ur data consists of a matrix with rows representing the individual cells, columns representing the expressed genes and values representing the gene expressions of each cell. The values of each row </span><span>range between</span><span class="c5">&nbsp;0 and 10 and there are no Nan values in our dataset. Because of the nature of our data we see no further need for preprocessing other than scaling. We use the StandardScaler algorithm for scaling. The StandardScaler algorithm standardizes the features of a dataset by subtracting the mean and dividing by the standard deviation. This normalization step is essential to ensure fair comparisons and optimal performance of machine learning models.</span></p><p class="c27 c26"><span class="c5"><br></span></p><h2 class="c29" id="h.scnm8v87nt4z"><span class="c19">Dimensionality Reduction</span></h2><h3 class="c37 c43" id="h.bmzkptaywdj"><span class="c7">PCA-UMAP-t-SNE </span></h3><p class="c35"><span class="c5">Dimension reduction algorithms can be classified into two categories based on their objectives: preserving the pairwise distance structure among all data samples or favoring the preservation of local distances over global distances. Algorithms such as PCA, MDS, and Sammon mapping fall into the former category. These methods aim to retain the overall relationships between data points. In contrast, t-SNE, UMAP, Laplacian eigenmaps, and diffusion maps belong to the latter category. These algorithms prioritize the preservation of local distance relationships.</span></p><p class="c15"><span class="c5">Principal Component Analysis (PCA) is a widely used dimensionality reduction technique based on singular values decomposition (SVD) that aims to capture the most informative aspects of high-dimensional data in a lower-dimensional representation.</span></p><p class="c15"><span>UMAP (Uniform Manifold Approximation and Projection) is a dimensionality reduction technique that has gained significant popularity in recent years for visualizing and analyzing high-dimensional data. UMAP aims to preserve both local and global data structure by constructing a low-dimensional representation of the data by modeling the data as a topological manifold. It has been shown to offer computational efficiency advantages over t-SNE. </span><span>However, </span><span class="c5">&nbsp;while UMAP outperforms t-SNE in terms of computational speed, it still faces challenges in the meaningful interpretation of inter cluster distances due to its use of a local neighbor graph approach.</span></p><p class="c15"><span>On the other hand, t-SNE is a popular nonlinear dimensionality reduction technique used for visualizing high-dimensional data in a lower-dimensional space. It was developed as an improvement over traditional techniques like PCA (Principal Component Analysis) </span><span>which</span><span class="c5">&nbsp; may not capture complex nonlinear relationships in the data. It constructs a low-dimensional representation of the data by modeling the similarity between data points in the original high-dimensional space and the lower-dimensional space. Nevertheless it has limitations regarding the preservation of global data structure and the meaningfulness of intercluster relations. It fails to preserve the overall structure of the data, making the intercluster relationships less interpretable. Additionally, t-SNE suffers from computational scalability issues when dealing with large datasets. </span></p><p class="c15"><span class="c5">Because of the lack of interpretability of their clustering results UMAP and t-SNE are mostly used for visualization purposes. </span></p><p class="c15"><span class="c5">In our analysis we use all the above mentioned algorithms with their default parameters for visualization of the dataset by selecting two and three components accordingly. Then try to optimize the PCA and the UMAP algorithms and keep their results for the downstream analysis. </span></p><p class="c1"><span class="c5"></span></p><h3 class="c42" id="h.1xamszrfpmhy"><span class="c7">2D and 3D visualization</span></h3><p class="c20"><span>Comparing the 2D and 3D plots we can certainly gain more insight from the 2D one, in which we can observe the forming of three distinct clusters. The above experiment is done just to get an insight </span><span>on</span><span class="c5">&nbsp; the structure of the data and it cannot be generalized to any dataset. We can also use this information to try and optimize our dimensionality reduction methods further.</span></p><p class="c27"><span><br></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 245.33px;"><img alt="" src="images/image10.png" style="width: 624.00px; height: 245.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c27"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 213.33px;"><img alt="" src="images/image2.png" style="width: 624.00px; height: 213.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><h3 class="c34 c40" id="h.lmhyqopgfjg7"><span class="c7"></span></h3><h3 class="c34" id="h.vol9joc5bvfd"><span class="c7">Optimizing UMAP using k-means clustering</span></h3><p class="c20"><span class="c5">If we want to optimize our first dimensionality reduction algorithm UMAP we will need an evaluation metric but the fact that we are working on an unsupervised learning problem gives us limited options. One option is to try to cluster the results of the dimensionality reduction algorithm and use the silhouette score as an evaluation metric. In the following blocks we will attempt to optimize UMAP using the silhouette score of a k-means clustering algorithm as a metric.The k-means algorithm is fast and robust to outliers but the price to pay for its assets is that the number of clusters k have to be defined prior to running the clustering. In our case we don&#39;t have any idea about the physical number of clusters that the data may form in the manifold representing them. Nevertheless we can make a rough guess based on the 2D representation we visualized on step one that the number of clusters will be 3. We can also identify the correct number of k by checking the results of our downstream analysis, namely the bic of our optimized clustering model, and select the one for which we observe a local minimum.</span></p><p class="c20"><span class="c5">We have to note here that the number k has to be defined by the user based on domain specific knowledge or prior visualization steps.</span></p><p class="c20"><span>We can visualize our data in 2 and 3 dimensions, although this visualization may not be enough to grasp the structure of the data if the resulting optimal number of components is greater than 3. Nevertheless we may get some insight</span><span>&nbsp;into</span><span class="c5">&nbsp;their structure.</span></p><p class="c27"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 550.00px; height: 458.00px;"><img alt="" src="images/image12.png" style="width: 550.00px; height: 458.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c27"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 558.67px;"><img alt="" src="images/image13.png" style="width: 624.00px; height: 558.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c27 c26"><span class="c5"><br></span></p><p class="c26 c27"><span class="c5"><br></span></p><p class="c27 c26"><span class="c5"><br></span></p><p class="c27 c26"><span class="c5"><br></span></p><p class="c27 c26"><span class="c5"><br></span></p><p class="c27 c26"><span class="c5"><br></span></p><h3 class="c34" id="h.qrzyuvh26txo"><span>Optimizing PCA using cumulative variance</span></h3><p class="c14"><span>In order to find the correct number of components to keep in our downstream analysis we will select the minimum PCA components that justify 98% of the variance of our data. Although we could get around with </span><span>fewer</span><span>&nbsp;components, thus simplifying the problem, we want to keep a number of components </span><span>that</span><span class="c5">&nbsp;justifies a big percentage of the variance of our data thus resulting in a more detailed differentiation between the clusters. It is though recommended to the user to experiment with different variance percentages thus keeping less components if there is a need of simplification or rather if there is no need for complexity.</span></p><p class="c27"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 574.00px; height: 458.00px;"><img alt="" src="images/image6.png" style="width: 574.00px; height: 458.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5"><br></span></p><p class="c27 c26"><span class="c5"></span></p><p class="c27 c26"><span class="c5"></span></p><p class="c27 c26"><span class="c5"></span></p><p class="c27 c26"><span class="c5"></span></p><h2 class="c41" id="h.ip1s8pv7zft8"><span class="c19">Clustering</span></h2><h3 class="c37 c43" id="h.ohf86kq0qg7r"><span class="c7">Optimize GMM model for PCA and UMAP results </span></h3><p class="c2"><span class="c13 c51"><br></span><span class="c5">We use Optuna to optimize the GMM algorithm. More specifically we want to select the best number of components between 2 and 10 and the best type of covariance matrix between &#39;full&#39;, &#39;tied&#39;, &#39;diag&#39;, &#39;spherical&#39;. We use a random initialization of our clusters and we evaluate the model&#39;s performance using BIC. We run Optuna&#39;s optimization function for 100 trials. We will run the algorithm for two separate datasets, the pca_data dataset which is the data results of the PCA reduction and the umap_data which is the data reduced by the UMAP algorithm.</span></p><p class="c11"><span class="c5"></span></p><p class="c14"><span class="c13 c48 c49">Optimised GMM model for the PCA dataset:</span></p><p class="c14"><span class="c39">Best number of components: 4</span></p><p class="c14"><span class="c39">Best covariance matrix type: full</span></p><p class="c14"><span class="c39">Best BIC score: 39852.79169417039</span></p><p class="c14 c26"><span class="c39"></span></p><p class="c14"><span class="c49 c13 c48">Optimised GMM model for the UMAP dataset:</span></p><p class="c14"><span class="c39">Best number of components: 3</span></p><p class="c14"><span class="c39">Best covariance matrix type: full</span></p><p class="c14"><span class="c39">Best BIC score: 981.0453925249503</span></p><h2 class="c12" id="h.mntxdlkbitsk"><span class="c19"></span></h2><p class="c11"><span class="c5"></span></p><p class="c11"><span class="c5"></span></p><h2 class="c12" id="h.n7beg4hs6pev"><span class="c19"></span></h2><h2 class="c12" id="h.k2bxgxel13ma"><span class="c19"></span></h2><h2 class="c43 c53" id="h.bzno4s8exycs"><span class="c19">Results</span></h2><p class="c2"><span class="c5">In the following section we present a visualization of our results by using scatter plots, heatmaps and boxplots in order to obtain the most information about our results.</span></p><h3 class="c37 c43" id="h.marn1487nnra"><span><br></span><span class="c7">GMM model for PCA Data</span></h3><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 569.50px; height: 452.68px;"><img alt="" src="images/image5.png" style="width: 569.50px; height: 452.68px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c5">We observe that we have four clearly separated clusters each one occupying a part of the four quadrants of the pca first two components plane. </span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 537.33px;"><img alt="" src="images/image7.png" style="width: 624.00px; height: 537.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c1"><span class="c17 c13"></span></p><p class="c2"><span class="c17 c13">Cell_joint_distributions </span></p><p class="c1"><span class="c17 c13"></span></p><a id="t.f0e632d18a0e780874ba7ad9b576ee163d996151"></a><a id="t.0"></a><table class="c45"><tr class="c31"><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c5">1</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td></tr><tr class="c31"><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">1</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td></tr><tr class="c31"><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">1</span></p></td><td class="c28" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0</span></p></td></tr></table><p class="c1"><span class="c17 c13"></span></p><p class="c2"><span>In the above heatmap and from our joint probability results we observe that each cell belongs exclusively to one of the 4 clusters although the sklearn&rsquo;s GMM uses a soft algorithmic scheme. The separability of our clusters indicates a good clustering model. </span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><h3 class="c37 c40" id="h.jdvqdqsply1x"><span class="c7"></span></h3><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 296.66px; height: 466.50px;"><img alt="" src="images/image1.png" style="width: 296.66px; height: 466.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 634.83px; height: 408.50px;"><img alt="" src="images/image11.png" style="width: 634.83px; height: 408.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c5">In the Barplot we observe the distribution of gene 1 in each one of the clusters and in the Heatmap we can observe the average expression of gene 1 in all of the four clusters. The gene expression to be depicted can be specified by the user by selecting the name of the plotted gene in the code.</span></p><h3 class="c37 c40" id="h.ih0px14620c1"><span class="c7"></span></h3><h3 class="c37 c40" id="h.lf43dc3qslbp"><span class="c7"></span></h3><h3 class="c37 c40" id="h.cdwihhjue9ye"><span class="c7"></span></h3><h3 class="c37" id="h.na7jg3pmonxv"><span>GMM model for UMAP Data</span></h3><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 497.33px;"><img alt="" src="images/image4.png" style="width: 624.00px; height: 497.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">In the scatterplot of the UMAP data clustering results we can once more observe three clearly separated clusters although we can see a point being misrepresented by cluster 1 while it seemingly belongs to cluster 0.</span></p><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 585.50px; height: 502.98px;"><img alt="" src="images/image9.png" style="width: 585.50px; height: 502.98px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c17 c13">Cell_joint_distributions </span></p><p class="c1"><span class="c5"></span></p><a id="t.dd7cd741235505b6743b3c96d3326fde324d49df"></a><a id="t.1"></a><table class="c45"><tr class="c31"><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">0.42083402</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0.</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c16"><span class="c17 c13">0.</span></p></td></tr><tr class="c31"><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">2.43329525e-313</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">3.76986293e-001</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">0.00000000e+000</span></p></td></tr><tr class="c31"><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">4.09003532e-235</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">&nbsp;0.00000000e+000</span></p></td><td class="c22" colspan="1" rowspan="1"><p class="c2"><span class="c17 c13">1.80632017e-001</span></p></td></tr></table><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c2"><span class="c5">By plotting the heatmap of each cell probability and by calculating the joint probability of the clusters we observe that each cell does not solely belong to a cluster. This is an indication of a less robust clustering result than the previous one. &nbsp;</span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 623.46px; height: 399.10px;"><img alt="" src="images/image8.png" style="width: 623.46px; height: 399.10px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><h1 class="c8 c44" id="h.2nnzg8z9dl5j"><span class="c32"></span></h1><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><p class="c2"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 388.49px; height: 584.50px;"><img alt="" src="images/image3.png" style="width: 388.49px; height: 584.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c5">Same as in the previous section in the Boxplot we observe the distribution of gene 1 in each of the clusters and in the Heatmap &nbsp;we can observe the average expression of gene 1 in all of the four clusters. The gene expression to be depicted can be specified by the user by selecting the name of the plotted gene in the code. In this way the user can search for the dominant genes of each cluster and specify the cell family each one belongs to. </span></p><h1 class="c8 c44" id="h.k0s5jefy18kl"><span class="c32"></span></h1><h1 class="c8" id="h.u9ojcp24j8nj"><span class="c32">Discussion</span></h1><p class="c2"><span>We utilized three dimensionality reduction algorithms for the visualization of our dataset. We attempted to optimize the UMAP algorithm utilizing </span><span>the</span><span>&nbsp;k-means algorithm. We kept the PCA components that explain 98% of the variance of our dataset. We performed two separate downstream </span><span>analyses</span><span class="c5">&nbsp;for our pca reduced data and our UMAP reduced data and compared their clustering results. We used GMM to cluster our data and optimized our model using Optuna. We observed that the GMM model for the UMAP reduced dataset achieved a better BIC score than the one for the pca reduced dataset. Nevertheless the pca reduced dataset achieved better cluster differentiation. In conclusion the BIC score is a relative measure and strongly depends on the dataset. We cannot compare the BIC scores of models deriving from different datasets. It&#39;s possible that the UMAP algorithm achieves a better BIC score because it creates a manifold where the data are better separated. However, because of the lack of symmetry conservation due to its non-linear nature we cannot achieve good clustering results in terms of separability. Thus it is possible that PCA is a more robust method for dimensionality reduction &nbsp;although we cannot conclude that without having explored UMAPs capabilities at their most. </span></p><p class="c1"><span class="c5"></span></p><p class="c14 c26"><span class="c9"></span></p><h2 class="c14 c43" id="h.4xzz08uujull"><span class="c19"><br>Bonus Research</span></h2><p class="c14"><span class="c5">The Agglomerative algorithm is widely used in scRNA sequence analysis thus we wanted to attempt to test its performance comparing it with the GMM model we created before. We followed the same dimensionality reduction steps and created our own pipeline optimizing an agglomerative algorithm. We evaluated the two pipelines generating our own data from a gaussian mixture distribution with a variable number of states. We evaluated the two pipelines using the Silhouette score, the Calinski-Harabasz Index and the Davies-Bouldin Index.</span></p><p class="c14"><span class="c5">The Calinski-Harabasz Index also known as the Variance Ratio Criterion Is a measure used to evaluate the quality of clustering results. It quantifies the ratio of between-cluster dispersion to within-cluster dispersion, aiming to identify clusters that are well-separated and compact.</span></p><p class="c14"><span class="c5"><br>The Davies-Bouldin score is a metric used to evaluate the quality of clustering results. It measures the average similarity between clusters and quantifies the trade-off between cluster cohesion (compactness) and cluster separation (distinctiveness).</span></p><p class="c14"><span class="c5">The Agglomerative algorithm outperformed the &nbsp;GMM model in all the three metrics. </span></p><p class="c14 c26"><span class="c5"></span></p><p class="c14 c26"><span class="c5"></span></p><h2 class="c41" id="h.1a5qe6g1y5tr"><span class="c19">Results</span></h2><p class="c11"><span class="c4"></span></p><p class="c11"><span class="c5"></span></p><p class="c24"><span class="c0">Number of States: 2</span></p><p class="c24"><span class="c0">Covariance Type: full</span></p><p class="c24"><span class="c0">Agglomerative Clustering Silhouette Score: 0.10142127464493851</span></p><p class="c24"><span class="c0">GMM Silhouette Score: -0.03130918707302906</span></p><p class="c24"><span class="c0">Agglomerative Clustering Calinski-Harabasz Index: 1.481959604964609</span></p><p class="c24"><span class="c0">GMM Calinski-Harabasz Index: 0.8752970705936518</span></p><p class="c24"><span class="c0">Agglomerative Clustering Davies-Bouldin score: 0.818172701113687</span></p><p class="c24"><span class="c0">GMM Davies-Bouldin score: 1.0645754191124766</span></p><p class="c11"><span class="c0"><br><br></span></p><p class="c24"><span class="c0">Number of States: 4</span></p><p class="c24"><span class="c0">Covariance Type: full</span></p><p class="c24"><span class="c0">Agglomerative Clustering Silhouette Score: 0.14687251436745075</span></p><p class="c24"><span class="c0">GMM Silhouette Score: -0.006682820884496267</span></p><p class="c24"><span class="c0">Agglomerative Clustering Calinski-Harabasz Index: 1.750810759285476</span></p><p class="c24"><span class="c0">GMM Calinski-Harabasz Index: 0.752118611191412</span></p><p class="c24"><span class="c0">Agglomerative Clustering Davies-Bouldin score: 0.7403624693662356</span></p><p class="c24"><span class="c0">GMM Davies-Bouldin score: 4.225147383734031</span></p><p class="c11"><span class="c5"></span></p><p class="c11"><span class="c5"></span></p><p class="c11"><span class="c5"><br></span></p><p class="c27 c26"><span class="c13 c17"></span></p><h1 class="c29" id="h.bjb2tc14l8xd"><span>References</span><span class="c32"><br></span></h1><p class="c27"><span>Preprossesing: </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html&amp;sa=D&amp;source=editors&amp;ust=1685741240576814&amp;usg=AOvVaw24c1qdSB2iiKUXz4jLLHQU">https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html</a></span></p><p class="c27"><span class="c5">Dimensionality Reduction:</span></p><p class="c27"><span>PCA</span><span class="c13">: </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html&amp;sa=D&amp;source=editors&amp;ust=1685741240577290&amp;usg=AOvVaw0yO8mM3PtMm9OTY0o6dnj1">https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html</a></span><span>, </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://en.wikipedia.org/wiki/Principal_component_analysis&amp;sa=D&amp;source=editors&amp;ust=1685741240577472&amp;usg=AOvVaw35zeF9DTGW4EyJU1Gy9IRd">https://en.wikipedia.org/wiki/Principal_component_analysis</a></span><span>, </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://towardsdatascience.com/how-to-select-the-best-number-of-principal-components-for-the-dataset-287e64b14c6d&amp;sa=D&amp;source=editors&amp;ust=1685741240577702&amp;usg=AOvVaw0BERth2aD8gyBf1dpz3Kvb">https://towardsdatascience.com/how-to-select-the-best-number-of-principal-components-for-the-dataset-287e64b14c6d</a></span><span class="c21">,</span><span><a class="c3" href="https://www.google.com/url?q=https://www.mdpi.com/1422-0067/21/16/5797&amp;sa=D&amp;source=editors&amp;ust=1685741240577869&amp;usg=AOvVaw0FylSAkrSz8V4kqBo2YRbB">&nbsp;</a></span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://www.mdpi.com/1422-0067/21/16/5797&amp;sa=D&amp;source=editors&amp;ust=1685741240578009&amp;usg=AOvVaw0pi851CoJfbdVfY6BrXHdj">https://www.mdpi.com/1422-0067/21/16/5797</a></span></p><p class="c27"><span>t-SNE :</span><span><a class="c3" href="https://www.google.com/url?q=https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html&amp;sa=D&amp;source=editors&amp;ust=1685741240578253&amp;usg=AOvVaw0vu-SPzl8OBzb6bxBUtGJJ">&nbsp;</a></span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html&amp;sa=D&amp;source=editors&amp;ust=1685741240578415&amp;usg=AOvVaw0FE-aw4rWpK5RIScP63-xT">https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html</a></span></p><p class="c27"><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://www.nature.com/articles/s41467-019-13056-x&amp;sa=D&amp;source=editors&amp;ust=1685741240578624&amp;usg=AOvVaw2iLAISfD8R42C75Z3GeyzY">https://www.nature.com/articles/s41467-019-13056-x</a></span><span class="c21">, </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://arxiv.org/abs/1802.03426&amp;sa=D&amp;source=editors&amp;ust=1685741240578796&amp;usg=AOvVaw24jaS07HWVPsSenhHRpoJg">https://arxiv.org/abs/1802.03426</a></span></p><p class="c27"><span>UMAP:</span><span><a class="c3" href="https://www.google.com/url?q=https://umap-learn.readthedocs.io/en/latest/parameters.html&amp;sa=D&amp;source=editors&amp;ust=1685741240579055&amp;usg=AOvVaw0jVEeVJNNJbfW1OzWuYNfr">&nbsp;</a></span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://umap-learn.readthedocs.io/en/latest/parameters.html&amp;sa=D&amp;source=editors&amp;ust=1685741240579208&amp;usg=AOvVaw05-7YLdNrsSOqnA75UiuR-">https://umap-learn.readthedocs.io/en/latest/parameters.html</a></span></p><p class="c27"><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://www.nature.com/articles/nbt.4314&amp;sa=D&amp;source=editors&amp;ust=1685741240579406&amp;usg=AOvVaw2DaQ8mv--lKQLCf23MuIQP">https://www.nature.com/articles/nbt.4314</a></span></p><p class="c27"><span>GMM: </span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://scikit-learn.org/stable/modules/mixture.html%23gmm&amp;sa=D&amp;source=editors&amp;ust=1685741240579656&amp;usg=AOvVaw3GlbJiNtBzl5GNRn8kMtNu">https://scikit-learn.org/stable/modules/mixture.html#gmm</a></span></p><p class="c27"><span class="c5">scRNA Best Practises:</span></p><p class="c27"><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://www.sc-best-practices.org/preprocessing_visualization/dimensionality_reduction.html&amp;sa=D&amp;source=editors&amp;ust=1685741240579977&amp;usg=AOvVaw0bzwcMh1DSDLVau18D-Eyo">https://www.sc-best-practices.org/preprocessing_visualization/dimensionality_reduction.html</a></span></p><p class="c27"><span class="c5">Evaluation metrics:</span></p><p class="c27"><span>:</span><span class="c21"><a class="c3" href="https://www.google.com/url?q=https://towardsdatascience.com/silhouette-coefficient-validating-clustering-techniques-e976bb81d10c&amp;sa=D&amp;source=editors&amp;ust=1685741240580361&amp;usg=AOvVaw32oUVnVXwSYiRX26RoSaCi">https://towardsdatascience.com/silhouette-coefficient-validating-clustering-techniques-e976bb81d10c</a></span></p><p class="c27"><span class="c18">https://scikit-learn.org/stable/modules/generated/sklearn.metrics.calinski_harabasz_score.html<br></span></p><p class="c27 c26"><span class="c18"><br></span></p><p class="c1"><span class="c5"></span></p><p class="c1"><span class="c5"></span></p><div><p class="c26 c52"><span class="c5"></span></p></div></body></html>